## Menghitung Probabilitas Posterior Menggunakan Teorema Bayes

### Langkah 1: Import Library

Import library yang dibutuhkan, dalam kasus ini kita akan menggunakan NumPy.

In [1]:
import numpy as np

### Langkah 2: Menentukan Probabilitas Prior

Probabilitas prior adalah probabilitas awal bahwa seorang pasien memiliki penyakit sebelum hasil tes dilakukan. Dalam kasus ini, probabilitas prior ditentukan sebagai berikut:

$$ P(\text{penyakit}) = 0.01 $$

In [17]:
probabilitas_penyakit = 0.01

### Langkah 3: Menghitung Probabilitas Posterior

Untuk menghitung probabilitas posterior, kita menggunakan Teorema Bayes. Langkah-langkahnya adalah sebagai berikut:

#### 3.1. Menghitung Probabilitas Tes Positif pada Individu yang Benar-benar Sakit

Probabilitas tes positif pada individu yang benar-benar sakit adalah probabilitas bahwa tes akan memberikan hasil positif jika subjeknya sebenarnya memiliki penyakit. Dalam kasus ini, kita menggunakan nilai kesalahan negatif palsu, yang merupakan persentase dari orang yang sebenarnya sakit tetapi dites negatif.

Misalnya, dengan tingkat kesalahan negatif palsu sebesar `2%`, ini berarti jika ada `100` orang yang sebenarnya sakit, sekitar `2` orang akan mendapat hasil tes negatif. Oleh karena itu, probabilitas tes positif pada individu yang benar-benar sakit adalah `98%` atau `0.98`.

Probabilitas tes positif pada individu yang benar-benar sakit dapat dihitung menggunakan rumus:

$$P(\text{positif}|\text{sakit}) = 1 - \text{tingkat kesalahan negatif palsu}$$

In [13]:
tingkat_kesalahan_negatif_palsu = 0.02
prob_positif_sakit = 1 - tingkat_kesalahan_negatif_palsu
print("Probabilitas Tes Positif Sakit:", prob_positif_sakit)


Probabilitas Tes Positif Sakit: 0.98


#### 3.2. Menghitung Probabilitas Tes Negatif pada Individu yang Tidak Sakit

Probabilitas tes negatif pada individu yang tidak sakit adalah probabilitas bahwa tes akan memberikan hasil negatif jika subjeknya sebenarnya sehat. Dalam kasus ini, kita menggunakan nilai kesalahan positif palsu, yang merupakan persentase dari orang yang sehat tetapi dites positif.

Misalnya, dengan tingkat kesalahan positif palsu sebesar `5%`, ini berarti jika ada `100` orang yang sebenarnya sehat, sekitar `5` orang akan mendapat hasil tes positif. Oleh karena itu, probabilitas tes negatif pada individu yang tidak sakit adalah `95%` atau `0.95`.

Probabilitas tes negatif pada individu yang tidak sakit dapat dihitung menggunakan rumus:

$$P(\text{negatif}|\text{sehat}) = 1 - \text{tingkat kesalahan positif palsu}$$

In [11]:
tingkat_kesalahan_positif_palsu = 0.05
prob_negatif_sehat = 1 - tingkat_kesalahan_positif_palsu
print("Probabilitas Tes Negatif yang Tidak Sakit:", prob_negatif_sehat)


Probabilitas Tes Negatif yang Tidak Sakit: 0.95


#### 3.3. Menghitung Probabilitas Tes Positif Secara Keseluruhan dengan Hukum Total Probabilitas

Probabilitas tes positif secara keseluruhan adalah probabilitas bahwa tes akan memberikan hasil positif pada individu apa pun, terlepas dari apakah mereka sehat atau sakit. Ini dihitung dengan menggunakan hukum total probabilitas, yang menggabungkan probabilitas hasil positif dari individu yang sehat dan sakit.

Misalnya, jika kita memiliki populasi di mana `1%` menderita penyakit, maka `99%` sisanya dianggap sehat. Dengan tingkat kesalahan positif palsu sebesar `5%`, ini berarti sekitar `5%` dari `99%` individu sehat akan menghasilkan tes positif, sedangkan `98%` dari `1%` individu sakit akan menghasilkan tes positif. 

Oleh karena itu, probabilitas tes positif secara keseluruhan adalah hasil dari perkalian probabilitas individu sehat menghasilkan tes positif dan probabilitas individu sakit menghasilkan tes positif.

Probabilitas tes positif secara keseluruhan dapat dihitung dengan hukum total probabilitas:

$$P(\text{positif}) = P(\text{positif}|\text{sakit}) \times P(\text{sakit}) + P(\text{positif}|\text{sehat}) \times P(\text{sehat})$$

In [18]:
prob_sehat = 1 - probabilitas_penyakit
prob_positif = prob_positif_sakit * probabilitas_penyakit + (1 - prob_negatif_sehat) * prob_sehat
print("Probabilitas Tes Positif Secara Keseluruhan dengan Hukum Total Probabilitas:", prob_positif)


Probabilitas Tes Positif Secara Keseluruhan dengan Hukum Total Probabilitas: 0.05930000000000005


#### 3.4. Menggunakan Teorema Bayes untuk Menghitung Probabilitas Posterior

Menggunakan Teorema Bayes, kita dapat menghitung probabilitas posterior bahwa seorang pasien yang mendapat tes positif pada tes laboratorium juga benar-benar memiliki penyakit ini. Ini menggabungkan probabilitas prior (probabilitas awal bahwa pasien memiliki penyakit) dengan hasil tes positif yang baru saja diperoleh.

Probabilitas posterior dapat dihitung menggunakan Teorema Bayes:

$$P(H|X) = \frac{{P(X | H) \times P(H)}}{{P(X)}}$$

Di mana:
- `P(H|X)` adalah probabilitas posterior dari hipotesis \( H \) setelah diberikan data \( X \).
- `P(X | H)` adalah probabilitas likelihood dari data \( X \) jika hipotesis \( H \) benar.
- `P(H)` adalah probabilitas prior dari hipotesis \( H \).
- `P(X)` adalah probabilitas dari data \( X \) secara keseluruhan.

Rumus ini digunakan untuk memperbarui probabilitas prior dengan informasi baru yang diberikan oleh data, sehingga menghasilkan probabilitas posterior yang diperbarui.

### Hasil Perhitungan


- Probabilitas Tes Negatif yang Tidak Sakit: `0.95`
- Probabilitas Tes Positif Sakit: `0.98`
- Probabilitas Tes Positif Secara Keseluruhan dengan Hukum Total Probabilitas: `0.05930000000000005`

Kita akan menggunakan Teorema Bayes untuk menghitung probabilitas posterior bahwa seorang pasien yang mendapat tes positif pada tes laboratorium juga benar-benar memiliki penyakit ini. Teorema Bayes menyatakan bahwa:

$$P(\text{sakit}|\text{positif}) = \frac{{P(\text{positif}|\text{sakit}) \times P(\text{sakit})}}{{P(\text{positif})}}$$

Di mana:

- `P(sakit|positif)` adalah probabilitas posterior bahwa pasien memiliki penyakit jika mendapat tes positif.
- `P(positif|sakit)` adalah probabilitas tes positif pada individu yang benar-benar sakit `(0.98)`.
- `P(sakit)` adalah probabilitas prior bahwa pasien memiliki penyakit `(1% atau 0.01)`.
- `P(positif)` adalah probabilitas tes positif secara keseluruhan `(0.05930000000000005)`.


Kita dapat menghitung probabilitas posterior menggunakan rumus tersebut:

$$P(\text{sakit}|\text{positif}) = \frac{{0.98 \times 0.01}}{{0.05930000000000005}}$$

$$P(\text{sakit}|\text{positif}) \approx \frac{{0.0098}}{{0.05930000000000005}} \approx 0.16526$$

In [28]:
probabilitas_posterior = (prob_positif_sakit * probabilitas_penyakit) / prob_positif
print("Probabilitas Posterior:", probabilitas_posterior)

Probabilitas Posterior: 0.16526138279932534


Jadi, probabilitas posterior bahwa seorang pasien yang mendapat tes positif pada tes laboratorium juga benar-benar memiliki penyakit ini adalah sekitar `16.526%`. Ini berarti ada sekitar `16.526%` kemungkinan bahwa pasien yang mendapat tes positif sebenarnya memiliki penyakit yang diuji.

### Kesimpulan

Probabilitas posterior ini memberikan perkiraan seberapa mungkin seorang pasien memiliki penyakit berdasarkan hasil tes laboratorium yang positif. Informasi ini dapat membantu dokter dalam proses diagnosa penyakit dengan memberikan panduan tambahan dalam menetapkan diagnosis dan memilih langkah-langkah pengobatan yang sesuai. Semakin tinggi probabilitas posterior, semakin besar kemungkinan pasien memiliki penyakit tersebut, dan dokter mungkin perlu melakukan lebih banyak tes atau pengamatan untuk mengonfirmasi diagnosis.